test-data/unit/check-isinstance.test

```diff
@@ -1344,3 +1344,11 @@ def f(x: object) -> None:
     reveal_type(b)  # E: Revealed type is '__main__.A'
 [builtins fixtures/isinstance.pyi]
 [out]
+
+[case testIsInstanceWithUnknownType]
+from typing import *
+def f(x: Union[int, str], typ: type) -> None:
+    if isinstance(x, (typ, int)):
+        x + 1  # E: Unsupported operand types for + (likely involving Union)
```

Comment:
Add also an `else:` block and do a `reveal_type(x)` in that block. Add another `reveal_type(x)` just after the if statement.